artiq/applets/plot_hist.py

```diff
@@ -24,10 +28,19 @@ def data_changed(self, data, mods, title):
             x = list(range(len(y)+1))
 
         if len(y) and len(x) == len(y) + 1:
+            self.timer.stop()
             self.clear()
             self.plot(x, y, stepMode=True, fillLevel=0,
                       brush=(0, 0, 255, 150))
             self.setTitle(title)
+        else:
+            self.timer.start(1000)
+
+    def length_warning(self):
+        self.clear()
+        text = "⚠️ dataset lengths mismatch:\n"\
+               "Bin boundaries should be one more than Y values"
```

Comment:
I think the correct grammar would be:
"There should be one more bin boundary than there are Y values."